pf/tfgen/main.go

```diff
@@ -235,22 +224,24 @@ func MainWithMuxer(provider string, infos ...tfbridge.Muxed) {
 			"Must provide a metadata store when muxing providers. See ProviderInfo.MetadataInfo")
 		err = metadata.Set(muxedInfo.GetMetadata(), "mux", mapping)
 		if err != nil {
-			return fmt.Errorf("Failed to save metadata: %w", err)
+			return fmt.Errorf("[pf/tfgen] Failed to set mux data in MetadataInfo: %w", err)
+		}
+
+		// In the muxing case precompute renames by merging them and set them to MetadataInfo, this will avoid
+		// recomputing the renames in GenerateFromSchema, it will write out the mergedRenames as-is.
+		mergedRenames := mergeRenames(renames)
```

Comment:
This bit is a bit odd. We'll be calling g.GenerateFromSchema below but we do not want that to recompute renames. We instead want it to accept the merged renames from all the tfgen.GenerateSchemaWithOptions calls done before this line is hit. So we set it upfront.

Reply:
I agree that we need to pull out renames here. I think we need to set rename somewhere that `GenerateFromSchema` is aware of, maybe with a `opts.PrecomputedRenames = mergeRenames(renames)`. 

See my other comment for why existence checking isn't sufficient.